test/util.ts

```diff
@@ -8,3 +9,14 @@ import {Spec} from '../src/schema/schema';
 export function parseModel(spec) {
   return new Model(spec as Spec);
 }
+
+export const zSchema = require('z-schema');
+
+zSchema.registerFormat('color', function (str) {
+  // valid colors are in list or hex color
+  return contains(['purple'], str) || /^#([0-9a-f]{3}){1,2}$/i.test(str);
```

Comment:
Cause that's the only color we actually use :-P Feel free to add it. I don't think we need to be exhaustive.